bundles/org.eclipse.search/search/org/eclipse/search/internal/ui/text/DirtyFileSearchParticipant.java

```diff
@@ -38,6 +40,7 @@
 
 import org.eclipse.search.internal.core.text.DirtyFileProvider;
 
+@Component(service = DirtyFileProvider.class, immediate = false, property = { "weight:Integer=100" })
```

Comment:
Maybe replace weight with ServiceRank?

Reply:
Actually it would have been a good choice when this service was introduced. But now that it is defined as property to sort difference service implementations I would consider this an API and for now I don't want to implement support for both in `DirtyFileSearchParticipantServiceTracker`.

https://github.com/eclipse-platform/eclipse.platform.ui/blob/04dc50b8b643a72416cb1779741827e36e9d2066/bundles/org.eclipse.search.core/search/org/eclipse/search/internal/core/DirtyFileSearchParticipantServiceTracker.java#L36